### PR TITLE
python3Packages.lmfit: init at 1.2.1

### PR DESCRIPTION
**Pull request by nomeata**
###### Description of changes

Adds a new python package. Yay, more packages!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable: Unit tests run, basic usage works.
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nomeata** [comment]
Upstream confirms that the skipped test cases are ok to skip: https://github.com/lmfit/lmfit-py/issues/878

File: pkgs/development/python-modules/lmfit/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  version = "1.2.1";
+
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-AM71vRRb+BtzYwt4kmrySyTxgFQh5iEcpYVYiqfMQVs=";
+  };
+
+
```
Comment (onny) [inline pkgs/development/python-modules/lmfit/default.nix:17]:
```suggestion
```

File: pkgs/development/python-modules/lmfit/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{ lib, buildPythonPackage, fetchPypi,
+  setuptools-scm, asteval, numpy, scipy, uncertainties,
+  pytest-cov, pytestCheckHook, pandas, matplotlib,
+}:
```
Comment (mweinelt) [inline pkgs/development/python-modules/lmfit/default.nix:4]: Please unpack these to one item per line.

File: pkgs/development/python-modules/lmfit/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+buildPythonPackage rec {
+  pname = "lmfit";
+  version = "1.2.1";
+
+  format = "setuptools";
```
Comment (mweinelt) [inline pkgs/development/python-modules/lmfit/default.nix:10]: Slight preference towards pyproject build, as it is more future-proof.

Requires `setuptools` and `setuptools-scm` in `nativeBuildInputs`. Not `wheel`, putting it in there is cargo culting.

File: pkgs/development/python-modules/lmfit/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  propagatedBuildInputs = [
+    setuptools-scm
+    asteval
+    numpy
+    scipy
+    uncertainties
+  ];
+
+  nativeCheckInputs = [
+    pytest-cov
```
Comment (mweinelt) [inline pkgs/development/python-modules/lmfit/default.nix:26]: We generally don't run coverage related tests. Remove this package and substitute the --cov argument here: https://github.com/lmfit/lmfit-py/blob/master/setup.cfg#L99

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by nomeata** [comment]
Thanks for the review and suggestions!

**Comment by onny** [comment]
Result of `nixpkgs-review pr 230865` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.lmfit</li>
    <li>python310Packages.lmfit.dist</li>
    <li>python311Packages.lmfit</li>
    <li>python311Packages.lmfit.dist</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Pull request merged**